G1ANT.Addon.UI/G1ANT.Addon.UI/Panels/UIControlsPanel.cs

```diff
@@ -121,5 +134,130 @@ private void refreshButton_Click(object sender, EventArgs e)
         {
             InitRootElement();
         }
+
+        #region RectangleForm
+
+        /// <summary>
+        /// Retrieves the top-level window that contains the specified UI Automation element.
+        /// </summary>
+        /// <param name="element">The contained element.</param>
+        /// <returns>The containing top-level window element.</returns>
+        private AutomationElement GetTopLevelWindow(AutomationElement element)
+        {
+            TreeWalker walker = TreeWalker.ControlViewWalker;
+            AutomationElement elementParent;
+            AutomationElement node = element;
+            do
+            {
+                elementParent = walker.GetParent(node);
+                if (elementParent == AutomationElement.RootElement) break;
+                node = elementParent;
+            }
+            while (true);
+            return node;
+        }
+
+        private void highlightToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                if (controlsTree.SelectedNode != null)
+                {
+                    if (controlsTree.SelectedNode.Tag is AutomationElement automationElement)
+                    {
+                        UIElement uiELement = new UIElement(automationElement);
+                        var element = UIElement.FromWPath(uiELement.ToWPath());
+                        if (element != null)
+                        {
+                            var window = GetTopLevelWindow(automationElement);
+                            if (window != null)
```

Comment:
What should happen when window is hidden or null? Don't highlight control

Reply:
done